src/processSymbolLayer.ts

```diff
@@ -93,31 +93,56 @@ const processSymbolLayerWithSubSymbol = (
   }
   if (symbol.type === "CIMLineSymbol") {
     if (layer.type === "CIMCharacterMarker") {
-      if (orientedMarkerAtStartOfLine(layer.markerPlacement)) {
-        const startSymbolizer = processOrientedMarkerAtEndOfLine(
-          layer,
-          "start",
-          options,
-        );
-        if (startSymbolizer) {
-          symbolizers.push(startSymbolizer);
+      const orientedMarkerPosition = getOrientedMarkerLinePosition(
+        layer.markerPlacement,
+      );
+      if (orientedMarkerPosition) {
+        switch (orientedMarkerPosition) {
+          case "start":
+            const startSymbolizer = processOrientedMarkerAtEndOfLine(
+              layer,
+              "start",
+              options,
+            );
+            if (startSymbolizer) {
+              symbolizers.push(startSymbolizer);
+            }
+            break;
+          case "end":
+            const endSymbolizer = processOrientedMarkerAtEndOfLine(
+              layer,
+              "end",
+              options,
+            );
+            if (endSymbolizer) {
+              symbolizers.push(endSymbolizer);
+            }
+            break;
+          case "both":
+            const startbothSymbolizer = processOrientedMarkerAtEndOfLine(
+              layer,
+              "start",
+              options,
+            );
+            const endbothSymbolizer = processOrientedMarkerAtEndOfLine(
+              layer,
+              "end",
+              options,
+            );
+            if (startbothSymbolizer && endbothSymbolizer) {
+              symbolizers.push(startbothSymbolizer);
+              symbolizers.push(endbothSymbolizer);
+            }
+            break;
```

Comment:
That looks working, but I think we can write this `switch` in a more compact way (which is great for maintenance and understandability):

```suggestion
        const processOrientedMarkerAtEndOfLineFn = (position: "start"|"end") => {
            const symbolizer = processOrientedMarkerAtEndOfLine(
              layer,
              position,
              options,
            );
            if (symbolizer) {
                symbolizers.push(symbolizer)
            }
        }
        if (orientedMarkerPosition === "start" || orientedMarkerPosition === "both")
            processOrientedMarkerAtEndOfLineFn("start");
        }
        if (orientedMarkerPosition === "end" || orientedMarkerPosition === "both")
            processOrientedMarkerAtEndOfLineFn("end");
        }
```